src/seedu/addressbook/data/AddressBook.java

```diff
@@ -143,4 +143,12 @@ public UniquePersonList getAllPersons() {
     public UniqueTagList getAllTags() {
         return new UniqueTagList(allTags);
     }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof AddressBook // instanceof handles nulls
+                && this.allPersons.equals(((AddressBook) other).allPersons)
+                && this.allTags.equals(((AddressBook) other).allTags));
```

Comment:
These two lines should be indented further because it is misleading to align `||` with `&&` (they are not at the same level in the condition tree)